lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -489,16 +489,28 @@ def commit(action, post, options, authorization = nil, method = :post)
 
         succeeded = success_from(action, response)
 
-        response(action, succeeded, response, options, source_id)
+        response(action, succeeded, response, options)
       end
 
-      def response(action, succeeded, response, options = {}, source_id = nil)
+      def commit(action, post, options, authorization = nil, method = :post)
+        if @options[:secret_key] || access_token_valid?
+          peform_payin(action, post, options, authorization, method)
```

Comment:
What will happen if the token is valid when this is executed, but due to communications lag, it expires before the gateway receives it? It will not attempt to refresh the token, and will just fail with an auth failure, right?

Reply:
The token is valid for 60mins so if there's a communication lag it would be a few minutes at most but yes if it expires it would fail with an auth failure. But if the customer is retries the transaction we would refresh the token.

Reply:
To prevent transactions from being made very close to the expiration and therefore failing due to lag, I was thinking that we could, for example, subtract a minute or two from the expiration time (`expires`) and with this we prevent there from being a lower failure rate. What do you think?